File: api-aggregator/src/test/java/com/github/alexeysol/geekregime/apiaggregator/features/user/service/v1/userservice/FindAllUsersTest.java
package com.github.alexeysol.geekregime.apiaggregator.features.user.service.v1.userservice;

import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.http.HttpStatus;

import java.util.List;

import static com.github.tomakehurst.wiremock.client.WireMock.*;

public class FindAllUsersTest extends BaseUserServiceTest {
    @Test
    public void usersExist_whenFindAllUsers_thenReturnsUserResponseList() {
        long userId = 1L;
        long userId2 = 2L;

        ResponseDefinitionBuilder responseToReturn = aResponse()
            .withStatus(HttpStatus.OK.value())
            .withBodyFile(getJsonPath("getAllUsers", HttpStatus.OK));

        wireMockServer.stubFor(getApiMappingBuilder(API_USERS_PATH, responseToReturn));

        List<Long> ids = List.of(userId, userId2);
        var users = service.findAllUsers(ids);

        wireMockServer.verify(getRequestedFor(urlPathEqualTo(API_USERS_PATH))
            .withHeader("Content-Type", equalTo("application/json")));

        int expectedListSize = 2;
        Assertions.assertEquals(expectedListSize, users.size());
        Assertions.assertEquals(userId, users.get(0).getId());
        Assertions.assertEquals(userId2, users.get(1).getId());
    }
}

File: api-users/src/test/kotlin/com/github/alexeysol/geekregime/apiusers/service/v1/userservice/FindAllUsersTest.kt
package com.github.alexeysol.geekregime.apiusers.service.v1.userservice

import com.github.alexeysol.geekregime.apiusers.model.entity.User
import io.mockk.every
import io.mockk.verify
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageImpl

class FindAllUsersTest : BaseUserServiceTest() {
    @Test
    fun usersExist_whenFindAllUsers_thenReturnsUserPage() {
        val users = listOf(
            User(details = defaultDetails),
            User(details = defaultDetails)
        )
        val userPage: Page<User> = PageImpl(users, pageableStub, users.size.toLong())

        every { repository.findAllUsers(pageableStub) } returns userPage

        val result = service.findAllUsers(pageableStub)
        verify(exactly = 1) { repository.findAllUsers(pageableStub) }
        Assertions.assertEquals(userPage, result)
    }

    @Test
    fun usersDontExist_whenFindAllUsers_thenReturnsEmptyPage() {
        val emptyUserPage: Page<User> = PageImpl(listOf(), pageableStub, 0)

        every { repository.findAllUsers(pageableStub) } returns emptyUserPage

        val result = service.findAllUsers(pageableStub)
        verify(exactly = 1) { repository.findAllUsers(pageableStub) }
        Assertions.assertEquals(emptyUserPage, result)
    }
}
